filemutex_flock.go

```diff
@@ -57,3 +58,15 @@ func (m *FileMutex) RUnlock() {
 	}
 	m.mu.RUnlock()
 }
+
+// Close does an Unlock() combined with closing and unlinking the associated
+// lock file. You should create a New() FileMutex for every Lock() attempt if
+// using Close().
+func (m *FileMutex) Close() {
+	if err := syscall.Flock(m.fd, syscall.LOCK_UN); err != nil {
+		panic(err)
+	}
+	syscall.Close(m.fd)
+	syscall.Unlink(m.path)
```

Comment:
I think there's no reason why anyone would want the lock file to exist if there will no longer be a lock held on it, so I don't think it makes sense to have 2 methods. So I could either rename Close() to CloseAndRemove(), or come up with some other shorter name that conveys things better. Finish()? This makes it clearer you're not supposed to use it again.